### ISSUE TITLE
Parse Sentinel jpeg2000 images

### ISSUE THREAD

**Issue by DanielJDufour** [GeoTIFF/georaster#20]
**Is your feature request related to a problem? Please describe.**

Need to parse in order to ease processing of Sentinel data 

**Describe the solution you'd like**

Same functions to parse as other files

**Additional context**

https://www.sentinel-hub.com/faq/how-can-i-get-source-data-scene

https://sentinel-s2-l1c.s3.amazonaws.com/tiles/10/S/DG/2015/12/7/0

**Comment by rowanwins** [comment]
A visible file browser for sentinel images is available here
http://sentinel-s2-l1c.s3-website.eu-central-1.amazonaws.com/#tiles/55/J/CF/2018/7/25/0/

Im thinking this might require writing some node bindings to some C written codec (eg [openJPEG](http://www.openjpeg.org/)). Or there is an [existing repo](https://github.com/jvrousseau/openjpeg) by @jvrousseau (who I've seen in various geo github circles) so that might be a starting point.

Will try to do some digging if I get a chance

**Comment by DanielJDufour** [comment]
@rowanwins , thanks for the leads.  I took a look at the links you sent and did a little more research.  My current thinking is we should create a separate JPEG 2000 decoding and encoding library that wraps around an existing Open JPEG port.  This new library would sorta serve the same function as geotiff.js but for JPEG 2000.  We can then import this new library into georaster.

**Comment by DanielJDufour** [comment]
I created https://github.com/GeoTIFF/jpeg2000.  @rowanwins , let me know if you want to be added as a collaborator :-)

**Comment by rowanwins** [comment]
Yep that seems like a sensible approach @DanielJDufour and seems like that would fit with the intro sentence on this repo 
> Wrapper around Georeferenced Rasters like GeoTIFF, NetCDF, JPG, and PNG that provides a standard interface

The only other thing I wonder is down the track whether it makes sense to split out the format support into their own libs so that if people are using `georaster` they can decide which format they want to support (eg I might not be interested in netcdf and so it would just add bloat to my code base)

So perhaps api wise you might end up with something like
````
const georaster = require("georaster");
const gt= require("georaster-geotiff");
const j2000= require("georaster-j2000");

const gr = new georaster({
  formats: [gt, j2000]
})

fetch(url)
  .then(response => response.arrayBuffer() )
  .then(gr)
  .then(georaster => {
      console.log("georaster:", georaster);
  });
````

Anyway that's just an idea if we ever get this j2000 stuff off the ground :) And yep happy to added as a contributor.

**Comment by DanielJDufour** [comment]
@rowanwins , cool.  That's a great point and I honestly hadn't thought about it yet.  We definitely don't want to make georaster heavy.  Making it work for advanced users such as yourself is definitely a top priority.  And I definitely appreciate your advice based on your deeper experience with open source GIS projects :-)

Two of my other considerations informing my thinking is ease of use for beginners and maintainability.  My current thinking is to try to keep all the georaster code in one source code repository, but using webpack to create separate bundles for each format.  We could then publish each of those bundles as separate packages on npmjs.

My mind is definitely not made up though and happy to discuss more.
